imports/ui/unit-explorer/unit-explorer.jsx

```diff
@@ -12,62 +12,169 @@ import { setDrawerState } from '../general-actions'
+          </div>
+        ) : <div>
+          <div>
+            {administratedUnits.length === 0 ? (undefined) : (
+              <div className='bb b--black-10 bg-very-light-gray f6 fw5 mid-gray pv2 pl2'>
+                Units I Administrate
+              </div>
+            )
+            }
+            {
+              administratedUnits.map(({ id, name, description, metaData }) => (
+                <MenuItem key={id} innerDivStyle={{padding: 0}} onClick={() => this.handleUnitClicked(id)} >
+                  <div className='mt2 ph2 bb b--very-light-gray br1 w-100 flex items-center pa2'>
+                    <FontIcon className='material-icons' color='var(--semi-dark-gray)'>home</FontIcon>
+                    <div className='ml3 mv1 semi-dark-gray lh-copy flex-grow overflow-hidden'>
+                      <div className='ti1 ellipsis'>{metaData.displayName || name}</div>
+                      <div className='ti1 ellipsis silver'>{ metaData.moreInfo || description}&nbsp;</div>
+                    </div>
+                    { !value ? (
+                      <div
+                        onClick={evt => this.handleAddCaseClicked(evt, id)}
+                        className='f6 ellipsis ml3 pl1 mv1 bondi-blue fw5 no-shrink'
+                      >
+                        Add case
+                      </div>
+                    ) : (undefined)
+                    }
+                  </div>
+                </MenuItem>
+              ))}
+          </div>
+          <div>
+            {unitsInvolvedIn.length === 0 ? (undefined) : (
+              <div className='bb b--black-10 bg-very-light-gray f6 fw5 mid-gray pv2 pl2'>
+                Units Involved In
+              </div>
+            )
+            }
+            {
+              unitsInvolvedIn.map(({ id, name, description, metaData }) => (
+                <MenuItem key={id} innerDivStyle={{padding: 0}} onClick={() => this.handleUnitClicked(id)}>
+                  <div className='mt2 ph2 bb b--very-light-gray br1 w-100 flex items-center pa2'>
+                    <FontIcon className='material-icons' color='var(--semi-dark-gray)'>home</FontIcon>
+                    <div className='ml3 mv1 semi-dark-gray lh-copy flex-grow overflow-hidden'>
+                      <div className='ti1 ellipsis'>{metaData.displayName || name}</div>
+                      <div className='ti1 ellipsis silver'>{ metaData.moreInfo || description}&nbsp;</div>
+                    </div>
+                    { !value ? (
+                      <div
+                        onClick={evt => this.handleAddCaseClicked(evt, id)}
+                        className='f6 ellipsis ml3 pl1 mv1 bondi-blue fw5 no-shrink'
+                      >
+                        Add case
+                      </div>
+                    ) : (undefined)
+                    }
+                  </div>
+                </MenuItem>
+              ))}
+          </div>
+        </div>
+        }
+      </div>
+    )
+  }
+}
```

Comment:
"dispatch" is still passed down to the object. It shouldn't be. The handlers need to be on the level of the component that uses the redux "connect" wrapper (in this case- UnitExplorer). Please use the handler-by-property pattern used elsewhere. For more information about why this pattern is needed, please see https://code-cartoons.com/a-cartoon-intro-to-redux-3afb775501a6#04c6 (we don't follow it very strictly, but the general guide is there)

Reply:
@nbiton I try to figure out how to pass unit id to these handler functions when sending these functions as props to FilteredUnitsList from Unit Explorer 🤔🤔. I understand that handler functions need to be inside of Unit Explorer class. I also read that function can be passed as props from the smart component in the cartoon (thank you for sharing!). I am confused because handleUnitClicked(id), handleAddCaseClicked(evt, id) need unit id and they are called inside of the .map function.

Could you hint me to a handler-by-property example in Unee-T code? I will read more about redux 😅😅

Reply:
The handlers can be called with the unit id, it's not a problem to call an external handler with arguments.
As for `evt.stopPropagation()`, it should still be done in the child component (FilteredUnitsList). Pass only the `id` to the handler on the parent component. The rule of thumb here is that the parent component shouldn't care how the event was captured by the child's implementation, and doesn't need to know about "stopPropagation" for this handler- it's not its concern.
For multiple examples of a smart component passing down handlers as props to "dumb" components, see case.jsx

Reply:
Handlers now in Unit Explorer are passed down to filteredUnitsList. I tried to figure out how to seperate concern and indicate `stopPropagation` in a pass down function. 

I also refactored to reduce code repetition in `Units Inolved In` and `Units I Administrate`. 

I rebased master to reflect the latest upstream. as a side effect, all the previous commits are now being displayed.